drizzlepac/haputils/poller_utils.py

```diff
@@ -686,15 +798,35 @@ def parse_obset_tree(det_tree, log_level):
 
                 # Determine if this image is a Grism/Prism or a nominal direct exposure
                 is_grism = False
-                if prod_list[6].lower().find('g') != -1 or prod_list[6].lower().find('pr') != -1:
+                if (
+                    prod_list[6].lower().find("g") != -1
+                    or prod_list[6].lower().find("pr") != -1
+                ):
                     is_grism = True
                     filt_indx -= 1
-                    grism_sep_obj = GrismExposureProduct(prod_list[0], prod_list[1], prod_list[2],
-                                                         prod_list[3], prod_list[4], filename[1], prod_list[6],
-                                                         prod_list[7], log_level)
+                    grism_sep_obj = GrismExposureProduct(
```

Comment:
The "aperture_from_poller" has been added to the class for the GrismExposureProduct, but the parameter is missing here.  It needs to be added here and NOT removed from the class.

Reply:
Sorry. I might be a little confused. `prod_list[4]` is `aperture_from_poller` and should line up with the correct input.

Reply:
issue appears resolved.